packages/medusa/src/api-v2/utils/validate-body.ts

```diff
@@ -4,17 +4,12 @@ import { z, ZodError } from "zod"
 import { MedusaRequest, MedusaResponse } from "../../types/routing"
 
 export async function zodValidator<T>(
-  zodSchema: z.ZodObject<any, any>,
-  body: T,
-  config: { strict?: boolean } = { strict: true }
+  zodSchema: z.ZodObject<any, any> | z.ZodEffects<any, any>,
+  body: T
 ): Promise<z.ZodRawShape> {
   try {
-    let schema = zodSchema
-    if (config.strict) {
-      schema = schema.strict()
-    }
-
-    return await schema.parseAsync(body)
+    zodSchema
```

Comment:
question: What happened here? Is this intentional?

Reply:
@olivermrbl the schema can return one of two types - ZodType and ZodEffect. You can only apply `strict` on ZodType. ZodEffect is returned if you do a `.refine` on the schema, which is a must for any XOR logic or cross-data checks. Therefore we cannot have this config, as `strict()` would have to be applied before calling `refine`.

I also think we can always have strict schemas, and if not we can just do the nonStrict definition as part of the validator, than as a config. Let me know if you think we can handle this in some other way